From 8591eba702f52dae335601c997271786be50dff7 Mon Sep 17 00:00:00 2001
From: Kevin Yue <yuezk001@gmail.com>
Date: Sun, 14 Dec 2025 17:10:52 +0800
Subject: [PATCH 2/4] Update user agent for common headers in GlobalProtect
 authentication

---
 auth-globalprotect.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/auth-globalprotect.c b/auth-globalprotect.c
index 8d094273..27e41c2b 100644
--- a/auth-globalprotect.c
+++ b/auth-globalprotect.c
@@ -34,14 +34,7 @@ struct login_context {
 void gpst_common_headers(struct openconnect_info *vpninfo,
 			 struct oc_text_buf *buf)
 {
-	char *orig_ua = vpninfo->useragent;
-
-	/* XX: more recent servers don't appear to require this specific UA value,
-	 * but we don't have any good way to detect them.
-	 */
-	vpninfo->useragent = (char *)"PAN GlobalProtect";
 	http_common_headers(vpninfo, buf);
-	vpninfo->useragent = orig_ua;
 }
 
 /* Translate platform names (derived from AnyConnect) into the values
-- 
2.50.1 (Apple Git-155)

