From 3915a51c81831a1a9b087d82f0c760365682f23b Mon Sep 17 00:00:00 2001
From: Kevin Yue <yuezk001@gmail.com>
Date: Mon, 15 Dec 2025 19:20:44 +0800
Subject: [PATCH 4/4] Add client version and OS information to HIP script
 arguments

---
 gpst.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/gpst.c b/gpst.c
index 0373fce0..d04c42c2 100644
--- a/gpst.c
+++ b/gpst.c
@@ -1013,6 +1013,12 @@ static int run_hip_script(struct openconnect_info *vpninfo)
 			exit(1);
 
 		hip_argv[i++] = openconnect_utf8_to_legacy(vpninfo, vpninfo->csd_wrapper);
+		hip_argv[i++] = "--client-version";
+		hip_argv[i++] = openconnect_get_gp_app_version(vpninfo);
+		hip_argv[i++] = "--client-os";
+		hip_argv[i++] = gpst_os_name(vpninfo);
+		hip_argv[i++] = "--os-version";
+		hip_argv[i++] = openconnect_get_gp_os_version(vpninfo);
 		hip_argv[i++] = "--cookie";
 		hip_argv[i++] = vpninfo->cookie;
 		if (vpninfo->ip_info.addr) {
@@ -1025,8 +1031,6 @@ static int run_hip_script(struct openconnect_info *vpninfo)
 		}
 		hip_argv[i++] = "--md5";
 		hip_argv[i++] = vpninfo->csd_token;
-		hip_argv[i++] = "--client-os";
-		hip_argv[i++] = gpst_os_name(vpninfo);
 		hip_argv[i++] = NULL;
 
 		/* XX: Sending the above parameters as --long-options was a mistake that was
-- 
2.50.1 (Apple Git-155)

